### PR TITLE
OCM-3828: HTPasswd IDP unique username validation

### PR DESCRIPTION
**Pull request by sagidayan**

### CODE REVIEW

**Review by osherdp — APPROVED** [review]

File: provider/identityprovider/htpasswd.go [hunk 1]
```diff
@@ -86,3 +89,24 @@ func CreateHTPasswdIDPBuilder(ctx context.Context, state *HTPasswdIdentityProvid
...
+func uniqueUsernameValidator() validator.List {
+	return attrvalidators.NewListValidator("userlist unique username", func(ctx context.Context, req validator.ListRequest, resp *validator.ListResponse) {
+		usersList := req.ConfigValue
+		htusers := []HTPasswdUser{}
+		err := usersList.ElementsAs(ctx, &htusers, true)
+		if err != nil {
+			resp.Diagnostics.AddAttributeError(req.Path, "Invalid list conversion", "Failed to parse userlist")
+			return
+		}
+		usernames := make(map[string]bool)
```
Comment (nirarg) [inline provider/identityprovider/htpasswd.go:102]: I think better to use here array (not map) and for each name check if already exists in the array

Comment (nirarg) [reply provider/identityprovider/htpasswd.go:102]: Ok, never mind, this is good

Comment (sagidayan) [reply provider/identityprovider/htpasswd.go:102]: :)
I do think that O(1) in finding in this case is "less expensive" than O(n) of memory...

**Pull request merged**